integration/tests/access/unstaked_node_test.go

```diff
@@ -214,3 +182,54 @@ func UnstakedNetworkingKey() (crypto.PrivateKey, error) {
 	}
 	return utils.GenerateUnstakedNetworkingKey(unittest.SeedFixture(n))
 }
+
+// followerManager is a convenience wrapper around the consensus follower
+type followerManager struct {
+	follower    *consensus_follower.ConsensusFollowerImpl
+	blockIDChan chan flow.Identifier
+	t           *testing.T
+}
+
+func newFollowerManager(t *testing.T, follower consensus_follower.ConsensusFollower) (*followerManager, error) {
+	followerImpl, ok := follower.(*consensus_follower.ConsensusFollowerImpl)
+	if !ok {
+		return nil, fmt.Errorf("unexpected consensus follower implementation")
+	}
+	fm := &followerManager{
+		follower:    followerImpl,
+		blockIDChan: make(chan flow.Identifier, blockCount),
+		t:           t,
+	}
+	follower.AddOnBlockFinalizedConsumer(fm.onBlockFinalizedConsumer)
+	return fm, nil
+}
+
+func (fm *followerManager) startFollower(ctx context.Context) {
+	go func() {
+		fm.follower.Run(ctx)
+	}()
+	// get the underlying node builder
+	node := fm.follower.NodeBuilder
+	// wait for the follower to have completely started
+	unittest.RequireCloseBefore(fm.t, node.Ready(), 10*time.Second,
+		"timed out while waiting for consensus follower to start")
+}
+
+func (fm *followerManager) onBlockFinalizedConsumer(finalizedBlockID flow.Identifier) {
+	// push the finalized block ID to the blockIDChannel channel
+	fm.blockIDChan <- finalizedBlockID
+}
+
+// hasBlock checks if the underlying storage of the consensus follower has a block
+func (fm *followerManager) hasBlock(blockID flow.Identifier) (bool, error) {
```

Comment:
I think the semantics of this method is a little weird.

imo, a method called `hasBlock` sounds like it should just return a boolean indicating whether we have the block or not. In the test above, you completely ignore the first return value, which sort of indicates imo that we should change this method.

An alternative, is to instead define a method called `getBlock` which returns `(block, error)`.